django/db/models/fields/__init__.py

```diff
@@ -857,10 +863,42 @@ def _get_default(self):
                 return self.default
             return lambda: self.default
 
+        if self.has_db_default():
+            return lambda: DB_DEFAULT
+
         if not self.empty_strings_allowed or self.null and not connection.features.interprets_empty_strings_as_nulls:
             return return_None
         return str  # return empty string
 
+    def has_db_default(self):
+        """Return a boolean of whether this field has a database default value."""
+        return self.db_default is not NOT_PROVIDED
+
+    def db_default_sql(self, model, schema_editor):
```

Comment:
This bi-directionnaly couples the model layer with the schema editor and should be avoided, only `connection` should be necessary.

`model` is also already available through `self.model` assuming the field is bound otherwise it should not make a huge difference since no field references should be resolvable.